apps/frontend/src/pages/SingleRequestPage/SingleRequestPage.tsx

```diff
@@ -0,0 +1,43 @@
+import { ActionFunctionArgs, LoaderFunctionArgs, useLoaderData } from 'react-router';
+import { Request } from '@neighborhood/backend/src/types';
+import { redirect } from 'react-router-dom';
+import requestServices from '../../services/requests';
+import RequestDescBox from '../../components/RequestDescBox/RequestDescBox';
+import { FullRequestData, SingleRequestFormIntent } from '../../types';
+
+export async function loader({ params }: LoaderFunctionArgs): Promise<FullRequestData | null> {
+  const { id } = params;
+  if (typeof id !== 'string') throw new Error('Invalid request id.');
+  const request = await requestServices.getSingleRequest(+id);
+
+  return request;
+}
+
+export async function action({ params, request }: ActionFunctionArgs) {
+  const formData = await request.formData();
+  const requestId = Number(params.id);
+  const neighborhoodId = formData.get('neighborhoodId');
+  const intent = formData.get('intent') as SingleRequestFormIntent;
+
+  let response: Request | '' | null = null;
+
+  if (intent === 'delete-request') {
+    response = await requestServices.deleteRequest(requestId);
+    return redirect(`/neighborhoods/${neighborhoodId}`);
+  } else if (intent === 'close-request') {
+    response = await requestServices.closeRequest(requestId);
+  }
+
+  return response;
+}
+
+export default function SingleRequestPage() {
+  const request = useLoaderData() as FullRequestData | null;
+  if (!request) return <div>Error</div>;
```

Comment:
Shouldn't we redirect to the error page here?

Reply:
Oops yep, I just forgot that there.

Reply:
how can I do that actually?

Reply:
Now that I think about it, we don't need line 36 at all.
In the app.tsx module we have the errorElement that automatically catches all Errors/Exceptions.
![Screenshot from 2023-11-09 12-52-41](https://github.com/neighborhood-app/app/assets/40914437/3e1ae019-5d79-484d-8938-c738f9764275)

Trying to access a non-existing request brings me to the error page automatically.
![Screenshot from 2023-11-09 12-53-59](https://github.com/neighborhood-app/app/assets/40914437/00677cfd-9cd4-4d99-b300-2c0ec916c8ef)

Reply:
Yeah, I know it's not required, I only added it for type-checking purposes. Do you have another idea? I need to return a valid JSX element in case that the `request` is `null`, otherwise TS is complaining. I could always silence the complaint, but I don't think that's the best way to go about this.